app/reflexes/products_reflex.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+class ProductsReflex < ApplicationReflex
+  include Pagy::Backend
+
+  before_reflex :init_filters_params, :init_pagination_params
+
+  def fetch
+    fetch_and_render_products
+  end
+
+  def change_per_page
+    @per_page = element.value.to_i
+    @page = 1
```

Comment:
I would expect to see the same elements as before, just more or less. We would need the old values though.

```
position = old_page * old_per_page
@page = position / @per_page
@page = @page.clamp(1, @page)
```

Reply:
i'm not sure to follow you. Would you like to make me a suggestion?

Reply:
I think what Maikel's saying is that whenever you change the 'per page' amount, it will reset you back to page 1, which is not intuitive. 

So I think this code suggestion is to select the closest page that includes some/all of the records that were previously on-screen. Is that right @mkllnk ?

Reply:
Exactly, I would see the same first row of the results and the tail of the results is extended or reduced.

Reply:
> whenever you change the 'per page' amount, it will reset you back to page 1, which is not intuitive.

Oh, ok, I thought it was intuitive ... Ok, let's do this!

> I would see the same first row of the results and the tail of the results is extended or reduced.

Makes sense!

Reply:
i can't make it work. I've tested the above code for sure, + tested mine.
I don't think it's that easy (specially when `per_page` aren't multiples each others) and I've spent to much time trying to fix it. Maybe open an issue?

Reply:
Okay, let's leave it for now then.